cfme/infrastructure/networking.py

```diff
@@ -1,73 +1,118 @@
 import attr
 from navmazing import NavigateToAttribute
+from widgetastic.widget import Text
 from widgetastic.widget import View
 from widgetastic_patternfly import Dropdown
 
 from cfme.base.ui import BaseLoggedInPage
+from cfme.common import CustomButtonEventsMixin
+from cfme.common import Taggable
 from cfme.modeling.base import BaseCollection
 from cfme.modeling.base import BaseEntity
 from cfme.utils.appliance.implementations.ui import CFMENavigateStep
+from cfme.utils.appliance.implementations.ui import navigate_to
 from cfme.utils.appliance.implementations.ui import navigator
+from widgetastic_manageiq import Accordion
+from widgetastic_manageiq import BaseEntitiesView
 from widgetastic_manageiq import ItemsToolBarViewSelector
-from widgetastic_manageiq import PaginationPane
-from widgetastic_manageiq import Text
+from widgetastic_manageiq import ManageIQTree
+from widgetastic_manageiq import SummaryTable
 
 
-class InfraNetworking(BaseEntity):
-    _param_name = 'InfraNetworking'
+@attr.s
+class InfraSwitches(BaseEntity, Taggable, CustomButtonEventsMixin):
+    """ Model of an infrastructure cluster in cfme
+
+    Args:
+        name: Name of the switch.
+    """
+
+    name = attr.ib()
 
 
 @attr.s
-class InfraNetworkingCollection(BaseCollection):
+class InfraSwitchesCollection(BaseCollection):
     """Collection object for the :py:class:`cmfe.infrastructure.networking.InfraNetworking`."""
-    ENTITY = InfraNetworking
+
+    ENTITY = InfraSwitches
+
+    def all(self):
+        """List of all switch objects"""
+        view = navigate_to(self, "All")
+        return [self.instantiate(ent) for ent in view.entities.entity_names]
 
 
 class InfraNetworkingView(BaseLoggedInPage):
     """Base view for header and nav checking, navigatable views should inherit this"""
 
     @property
     def in_infra_networking(self):
-        nav_chain = ['Compute', 'Infrastructure', 'Networking']
-        return (
-            self.logged_in_as_current_user and
-            self.navigation.currently_selected == nav_chain)
+        nav_chain = ["Compute", "Infrastructure", "Networking"]
+        return self.logged_in_as_current_user and self.navigation.currently_selected == nav_chain
 
 
 class InfraNetworkingToolbar(View):
-    """The toolbar on the main page"""
-    policy = Dropdown('Policy')
+    """The toolbar for Infra Networking view"""
 
+    policy = Dropdown("Policy")
     view_selector = View.nested(ItemsToolBarViewSelector)
 
 
-class InfraNetworkingEntities(View):
-    """Entities on the main page"""
-    title = Text('//div[@id="main-content"]//h1')
+class InfraSwitchesAllView(InfraNetworkingView):
+    """Infra switch all view"""
+
+    @property
+    def is_displayed(self):
+        return self.in_infra_networking and self.entities.title.text == "All Switches"
+
+    toolbar = View.nested(InfraNetworkingToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
```

Comment:
is paginator not used in any other tests?
like it was `paginator = PaginationPane()`

Reply:
@lina-nikiforova that was old implementation. I remember, on your PR i added one suggestion.
Lina as I am using `BaseEntitiesView` its cover `paginator` for me.

Reference: https://github.com/ManageIQ/integration_tests/blob/722dfac39a54d7bf2ea5d52506175930df90d177/widgetastic_manageiq/__init__.py#L3070